mbed-drivers/test_env.h

```diff
@@ -23,79 +23,51 @@
-void notify_performance_coefficient(const char* measurement_name, const double value);
+void notify_timeout(const int);
+void notify_hosttest(const char *);
+void notify_completion(const int);
+void notify_testcase_start(const char *);
+void notify_testcase_finish(const char *, const int);
 
-// Host test auto-detection API
-void notify_host_test_name(const char *host_test);
-void notify_timeout(int timeout);
-void notify_test_id(const char *test_id);
-void notify_test_description(const char *description);
+// scanf replacement tokenizer and simple KV grammar parser
+int greentea_parse_kv(char *, char *, const int, const int);
 
+
+#ifdef YOTTA_CFG_DEBUG_OPTIONS_COVERAGE
 // Code Coverage API
 void notify_coverage_start(const char *path);
 void notify_coverage_end();
+#endif
 
-// Host test auto-detection API
-#define MBED_HOSTTEST_START(TESTID)      notify_test_id(TESTID); notify_start()
-#define MBED_HOSTTEST_SELECT(NAME)       notify_host_test_name(#NAME)
-#define MBED_HOSTTEST_TIMEOUT(SECONDS)   notify_timeout(SECONDS)
-#define MBED_HOSTTEST_DESCRIPTION(DESC)  notify_test_description(#DESC)
-#define MBED_HOSTTEST_RESULT(RESULT)     notify_completion(RESULT)
-#define MBED_HOSTTEST_ASSERT(cond)       \
-    do {                                 \
-        if (!(cond)) {                   \
-            printf("HOSTTEST ASSERTION FAILED: '%s' in %s, line %d\r\n", #cond, __FILE__, __LINE__); \
-            notify_completion(false);    \
-        }                                \
-    } while(false)
-
-/**
-    Test auto-detection preamble example:
-    main() {
-        MBED_HOSTTEST_TIMEOUT(10);
-        MBED_HOSTTEST_SELECT( host_test );
-        MBED_HOSTTEST_DESCRIPTION(Hello World);
-        MBED_HOSTTEST_START("MBED_10");
-        // Proper 'host_test.py' should take over supervising of this test
-
-        // Test code
-        bool result = ...;
-
-        MBED_HOSTTEST_RESULT(result);
-    }
-*/
-
-
-// Test functionality useful during testing
-unsigned int testenv_randseed();
 
-// Macros, unit test like to provide basic comparisons
-#define TESTENV_STRCMP(GIVEN,EXPECTED) (strcmp(GIVEN,EXPECTED) == 0)
+// Test suite Host Test auto-detection macros
 
-// macros passed via test suite
-#ifndef TEST_SUITE_TARGET_NAME
-#define TEST_SUITE_TARGET_NAME "Unknown"
-#endif
+#define GREENTEA_START()                            notify_start();
+#define GREENTEA_SETUP(TIMEOUT, HOST_TEST_NAME)     notify_timeout(TIMEOUT); notify_hosttest(HOST_TEST_NAME);
+#define GREENTEA_SEND_KV(KEY,VALUE)                 greentea_send_kv(KEY,VALUE);
 
-#ifndef TEST_SUITE_TEST_ID
-#define TEST_SUITE_TEST_ID "Unknown"
-#endif
-
-#ifndef TEST_SUITE_UUID
-#define TEST_SUITE_UUID "Unknown"
-#endif
+#define GREENTEA_TSUITE_RESULT(RESULT)              notify_completion(RESULT);
+#define GREENTEA_TCASE_START(TESTCASE_UD)           notify_testcase_start(TESTCASE_UD);
+#define GREENTEA_TCASE_FINISH(TESTCASE_UD,SUCCESS)  notify_testcase_finish(TESTCASE_UD,SUCCESS);
```

Comment:
Also, could we _please_ spare the three more letters and call this `GREENTEA_TESTCASE_*`?